### PR TITLE
feat(typescript): compile-time missing secret error

### PR DESCRIPTION
**Pull request by jablko**
Should this be a compile-time as well as a run-time error?
```Diff
--- a/test/unit/middleware-constructor-test.ts
+++ b/test/unit/middleware-constructor-test.ts
@@ -1,5 +1,6 @@
 import { createMiddleware as Middleware } from "../../src/middleware";
 
 test("options: none", () => {
+  // @ts-expect-error
   expect(() => Middleware({})).toThrow();
 });

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -42,12 +41,12 @@ class Webhooks<
...
   ) => Promise<void>;
 
-  constructor(options?: Options<E, TTransformed>) {
+  constructor(options?: Options<E, TTransformed> & { secret: string }) {
     if (!options || !options.secret) {
       throw new Error("[@octokit/webhooks] options.secret required");
     }
 
-    const state: State = {
+    const state = {
```
Comment (G-Rath) [inline src/index.ts]:
```suggestion
    const state: State & { secret: string }= {
```

This needs to be explicitly typed as it's an object

File: src/middleware/index.ts [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
+import { Options } from "../types";
 
-export function createMiddleware(options: Options<any>) {
+export function createMiddleware(options: Options<any> & { secret: string }) {
   if (!options || !options.secret) {
     throw new Error("[@octokit/webhooks] options.secret required");
   }
 
-  const state: State = {
+  const state = {
```
Comment (G-Rath) [inline src/middleware/index.ts]: This needs to be explicitly typed as its an object.

**Review by G-Rath — DISMISSED** [review]

File: src/index.ts (hunk 1 above)
Comment (jablko) [reply src/index.ts]: I've applied this change although I confess I don't understand it: why do objects need to be explicitly typed?

**Review by wolfy1339 — DISMISSED** [review]

File: src/index.ts (hunk 1 above)
Comment (jablko) [reply src/index.ts]: Answered in https://github.com/octokit/webhooks.js/pull/466#discussion_r575863299, thanks!

**Review by gr2m — APPROVED** [review]
Thanks! We shoudl have that a long time ago! I am sure we will be able to make the types much more elegant once we split out the event handling into its own module

**Pull request merged**